### ISSUE TITLE
be careful about the `unofficial client` check and report (in browser)

### ISSUE THREAD

**Issue by zixia** [Chatie/wechaty#29]
TL;DR

when we inject our hooking code to https://wx.qq.com , we should keep the environment as clean as possible.

---

there has javascript code in wx.qq.com, which will check the run environment to see if had been modified: for example, the checksum of the whole angularjs object.

when we inject some code to https://wx.qq.com , we should keep the environment as clean as possible. because if a user account is reported using an unofficial client, the system might add some extra limitation to this account, for example, this account can not "say hi" to other strangers. (as the following screen shot, I think it's purpose is to prevent spamming other users by robot)

![wechat-unofficial-client-warning](https://cloud.githubusercontent.com/assets/1361891/18390990/02bd3964-76df-11e6-88dd-53e82dd8708c.jpg)
![wechat-moments-disabled-hooltool](https://cloud.githubusercontent.com/assets/1361891/19836554/4bf22392-9edf-11e6-93d7-04ebe43aac14.jpg)

TODO
- [ ] disable API_webwxreport: "/cgi-bin/mmwebwx-bin/webwxstatreport"

### See

1. [揭秘“微信群控” - 微信安全中心](https://mp.weixin.qq.com/s?__biz=MzA5MzEzNDg3MQ==&mid=2652767733&idx=1&sn=566c96eff753319b9361fb7930dee065&chksm=8b882c8dbcffa59b35f1ce988d276777f3c75c040b4717a1166e5fe7bf93f06dd5f40f4bc912&mpshare=1&scene=1&srcid=06025VJqxq4r1xjlkb2bbH3l#rd)

**Comment by zixia** [comment]
Update: 
1. Tencent blocks lots of bots in Aug 2017. I believe it based on the behavior analytics of the bot, with a gray upgrade progress. See #603

**Comment by zixia** [comment]
One year passed, we had been able to do more with Wechaty than before: we can not only be able to control a Web Browser, but also we can use iPad protocol directly, and the iPhone App Hook Puppet was already in alpha stage and will be ready to use soon.

However, I believe all the above methods will not change the problem that this issue is addressed: Tencent will not like a program to control the Wechat account.

The reason for that is because, till today, almost all the bots are doing _bad_ things like sending mass messages for advertising, rob red package money from human, or even doing grift by the program automatically, which will cover huge users in a very short time.

Wechaty do not like those _bad things_ too. We want to do good things for the user, for example, by integrating with Conversational AI with Wechaty, we can build a bot that improve user experiences a lot, serve users more with powerful ChatBot technology, which the others technology gaint had already started to do this years before, like Facebook Messager, Google Assitant and Microsoft Skype.

I'll close this issue because it's outdated, but this topic will be very valuable for future discuss.
